Orange/widgets/visualize/owfreeviz.py

```diff
@@ -15,128 +14,48 @@
-    cancelled = Signal()
-
-    def __init__(self, parent=None, **kwargs):
-        super().__init__(parent, **kwargs)
-        self.__coroutine = None
-        self.__next_pending = False  # Flag for compressing scheduled events
-        self.__in_next = False
-        self.__state = AsyncUpdateLoop.Idle
-
-    @Slot(object)
-    def setCoroutine(self, loop):
-        """
-        Set the coroutine.
-
-        The coroutine will be resumed (repeatedly) from the event queue.
-        If there is an existing coroutine set it is first closed/cancelled.
-
-        Raises an RuntimeError if the current coroutine is running.
-        """
-        if self.__coroutine is not None:
-            self.__coroutine.close()
-            self.__coroutine = None
-            self.__state = AsyncUpdateLoop.Cancelled
-
-            self.cancelled.emit()
-            self.finished.emit()
-
-        if loop is not None:
-            self.__coroutine = loop
-            self.__state = AsyncUpdateLoop.Running
-            self.__schedule_next()
-
-    @Slot()
-    def cancel(self):
-        """
-        Cancel/close the current coroutine.
-
-        Raises an RuntimeError if the current coroutine is running.
-        """
-        self.setCoroutine(None)
-
-    def state(self):
-        """
-        Return the current state.
-        """
-        return self.__state
-
-    def isRunning(self):
-        return self.__state == AsyncUpdateLoop.Running
-
-    def __schedule_next(self):
-        if not self.__next_pending:
-            self.__next_pending = True
-            QTimer.singleShot(10, self.__on_timeout)
-
-    def __next(self):
-        if self.__coroutine is not None:
-            try:
-                rval = next(self.__coroutine)
-            except StopIteration as stop:
-                self.__state = AsyncUpdateLoop.Finished
-                self.returned.emit(stop.value)
-                self.finished.emit()
-                self.__coroutine = None
-            except BaseException as er:
-                self.__state = AsyncUpdateLoop.Finished
-                self.raised.emit(er)
-                self.finished.emit()
-                self.__coroutine = None
-            else:
-                self.yielded.emit(rval)
-                self.__schedule_next()
-
-    @Slot()
-    def __on_timeout(self):
-        assert self.__next_pending
-        self.__next_pending = False
-        if not self.__in_next:
-            self.__in_next = True
-            try:
-                self.__next()
-            finally:
-                self.__in_next = False
-        else:
-            # warn
-            self.__schedule_next()
+class Results(namespace):
+    projector = None  # type: FreeViz
+    projection = None  # type: FreeVizModel
 
-    def customEvent(self, event):
-        if event.type() == AsyncUpdateLoop.Next:
-            self.__on_timeout()
-        else:
-            super().customEvent(event)
+
+class Runner:
```

Comment:
Like I wrote for the example widget, there's no need for this to be wrapped in a class. Just rename it to something like `run_freeviz` and have it be a separate function.